projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts

```diff
@@ -39,8 +39,9 @@ export class RelatedDiscussionComponent implements OnInit, OnChanges {
     this.discussionService.fetchSingleCategoryDetails(cid).subscribe(
       (data: NSDiscussData.ICategoryData) => {
         this.relatedDiscussions = [];
+        // filter to get the not deleted topics
         _.filter(data.topics, (topic) => {
-          if (this.topicId != topic.tid) {
+          if (topic.deleted == 0 && this.topicId != topic.tid) {
```

Comment:
Why this condition is added ?

Reply:
This condition has been added to get only the discussions which are not deleted from nodebb